### DISCUSSION TITLE
Cherry pick with --strategy and -X option

### DISCUSSION THREAD

**Discussion by noorul** [magit/magit#4726] [Support] [answered]
How do I run the following command using transient?

```bash
git cherry-pick  --strategy=recursive -X theirs 027eb01 7b29437 9f0d1a9 87b08f8 
```

**Comment by kyleam** [comment, marked as answer]
> --strategy=recursive

This part is available with `=s`.

> -X theirs

This isn't currently part of the `magit-cherry-pick` transient, though you [could add it yourself](https://magit.vc/manual/transient/Modifying-Existing-Transients.html#Modifying-Existing-Transients).  For example:

```
(transient-insert-suffix 'magit-cherry-pick "=s"
  '(magit-merge:--strategy-option))
```

> 027eb01 7b29437 9f0d1a9 87b08f8

If this is a string of commits that can be displayed in a section (e.g., the log buffer), then you can select them with the region before invoking the `magit-cherry-pick` transient.  However, the more general case of specifying an arbitrary set of commits isn't supported by Magit.

**Comment by noorul** [comment]
Thank you @kyleam for the detailed answer. So currently passing -X is not supported out of the box.
